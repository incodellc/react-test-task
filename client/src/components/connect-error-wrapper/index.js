import ConnectErrorWrapper from './ConnectErrorWrapper';
export default ConnectErrorWrapper;