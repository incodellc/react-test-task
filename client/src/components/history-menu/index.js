import HistoryMenu from './HistoryMenu';
export default HistoryMenu;