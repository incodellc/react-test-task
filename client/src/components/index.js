import InfoBar from './info-bar';
import Layout from './layout';
import ConnectErrorWrapper from './connect-error-wrapper';
import TickerList from './ticker-list';
import TickerListItem from './ticker-list-item';

export { InfoBar, Layout, ConnectErrorWrapper, TickerList, TickerListItem };