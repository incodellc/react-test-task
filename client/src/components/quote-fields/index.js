import QuoteFields from './QuoteFields';
export default QuoteFields;