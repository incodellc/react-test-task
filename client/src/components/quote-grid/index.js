import QuotesGrid from './QuotesGrid';
export default QuotesGrid;