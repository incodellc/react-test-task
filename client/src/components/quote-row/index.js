import QuoteRow from './QuoteRow';
export default QuoteRow;