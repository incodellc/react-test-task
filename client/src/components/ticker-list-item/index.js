import TickerListItem from './TickerListItem';
export default TickerListItem;