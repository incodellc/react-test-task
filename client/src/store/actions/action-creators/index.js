import actionTypes from '../action-types';
import { tickerService } from '../../../services';
import { tickers } from '../../../utils';

export const streamServerConnect = () => {
    return {
        type: actionTypes.STREAM_SERVER_CONNECT
    };
};

export const streamServerDisconnect = () => {
    return {
        type: actionTypes.STREAM_SERVER_DISCONNECT
    };
};

export const updateQuotes = (data) => {
    return {
        type: actionTypes.UPDATE_QUOTES,
        data
    };
};

export const connectToStreamServer = () => {
    return dispatch => {
        dispatch({
            type: actionTypes.CONNECTION_INIT
        });
        tickerService(
            tickers, 
            () => dispatch(streamServerConnect()), 
            (data) => dispatch(updateQuotes(data)),
            () => dispatch(streamServerDisconnect())
        );
    };
};