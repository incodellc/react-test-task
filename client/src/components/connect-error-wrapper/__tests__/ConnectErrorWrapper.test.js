import React from 'react';
import { renderWithRedux } from '../../../utils/test-utils';
import { initState } from '../../../store/reducer';
import { connectToStreamServer } from '../../../store/actions/action-creators';
import { ConnectErrorWrapper } from '../../../components';
import { cleanup, fireEvent } from '@testing-library/react';
jest.mock('../../../services');

describe('Check App component', () => {
    afterEach(cleanup);

    it('component renders and behave correctly', () => {
        const { asFragment, queryByTestId, getByText, store } = renderWithRedux(
            <ConnectErrorWrapper 
                onReconnectClick={() => store.dispatch(connectToStreamServer())}
            >
                    <div>Content</div>
            </ConnectErrorWrapper>,
            { ...initState, connecting: false }
        );
        expect(asFragment()).toMatchSnapshot();
        expect(queryByTestId('connect-error-gag')).toBeInTheDocument();
        expect(queryByTestId('reconnect-button')).toBeInTheDocument();
        expect(queryByTestId('connect-error-gag')).toHaveTextContent('Ð¡onnection failed.');
        expect(queryByTestId('reconnect-button')).toHaveTextContent('Reconnect');
        fireEvent.click(queryByTestId('reconnect-button'));
        expect(queryByTestId('connect-error-gag')).not.toBeInTheDocument();
        expect(queryByTestId('reconnect-button')).not.toBeInTheDocument();
        expect(getByText('Content')).toBeInTheDocument();
    });
});