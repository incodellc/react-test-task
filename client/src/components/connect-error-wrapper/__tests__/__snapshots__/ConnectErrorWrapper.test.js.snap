// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check App component component renders and behave correctly 1`] = `
<DocumentFragment>
  <div
    class="ui basic placeholder segment gag"
    data-testid="connect-error-gag"
  >
    <div
      class="ui icon header"
    >
      <i
        aria-hidden="true"
        class="frown outline icon"
      />
      Ð¡onnection failed.
    </div>
    <button
      class="ui negative button"
      data-testid="reconnect-button"
    >
      Reconnect
    </button>
  </div>
</DocumentFragment>
`;
